ibis/backends/base/sql/alchemy/datatypes.py

```diff
@@ -30,15 +30,15 @@ def compiles_array(element, compiler, **kw):
 class StructType(sat.UserDefinedType):
     def __init__(
         self,
-        pairs: Iterable[tuple[str, sat.TypeEngine]],
+        pairs: Iterable[tuple[str, sat.TypeEngine]] | Mapping[str, sat.TypeEngine],
     ):
-        self.pairs = [(name, sat.to_instance(type)) for name, type in pairs]
+        self.pairs = {name: sat.to_instance(type) for name, type in dict(pairs).items()}
```

Comment:
I find pairs a bit misleading, perhaps naming it as `fields` would be slightly better.

Reply:
Roger, fixing.